include/revng-c/Support/Annotations.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/ADT/ConstexprString.h"
+#include "revng/ADT/STLExtras.h"
+
+namespace ptml {
+
+// Attribute in this context is a revng-specific macro that gets unrolled into
+// an `__attribute__(($something))` for the compiler.
+struct Attribute {
+  std::string_view Macro;
+  std::string_view Value;
+  bool IsReal = false;
+};
+
+// Annotation is an attribute that can also encode a value
+// (for example an abi name).
+struct Annotation {
+  std::string_view Macro;
+  std::string_view Prefix;
+  std::string_view Suffix;
```

Comment:
It seems to me that this is unused.

Not even the `getDefinitions` function uses this.

Reply:
Let's just drop this for now.

Reply:
Ah, whoops, thanks for noticing. It was a left-over. In an earlier implementation, prefix also contained `annotate("`, which required `")` suffix. But that proven to bring more headaches that it solved. But it seems i forgot to remove the suffix :D

Reply:
Gone.